### ISSUE TITLE
missing master role in ganeti 2.9.X

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#743]
Originally reported of Google Code with ID 687.

```
(reported by George Kargiotakis via email)

<b>What steps will reproduce the problem?</b>
1. run 'gnt-node list -o role'

<b>What is the expected output? What do you see instead?</b>

Expected to see precisely one master node, found none; in fact, the master node was reported as master candidate.

<b>Please use labels and text to provide additional information.</b>

Dimara proposed the attached patch.
```
Originally added on 2014-01-20 10:26:01 +0000 UTC.

Attachments: 
[masterrole.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-687/comment-0/masterrole.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
sorry, it was Appolon who propsed the patch
```
Originally added on 2014-01-20 10:27:03 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The patch fixes a valid problem (so pushed. Thanks!), but unfortunately not the 'gnt-node list -o role' issue, as 'gnt-node list' uses the query socket. Will work on a fix in luxid, too.
```
Originally added on 2014-01-20 12:10:26 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 368e95fd8412e25e87c14699059a701a15ae7bdf
Author: Apollon Oikonomopoulos <apoikos@gmail.com>
Date:   Mon Jan 20 14:55:26 2014 +0200

    luxid: fix detection of master node in node query
    
    Ganeti.Config.getNodeRole would rely on clusterMasterNode returning the
    master node name, however clusterMasterNode returns the master node's
    UUID. We fix this and a similar issue in Ganeti.Query.Node.nodeFields.
    
    Together with 1ec34e26, this fixes issue #687.
    
    Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

```
Originally added on 2014-01-20 13:06:52 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-01-27 15:49:41 +0000 UTC.

Changed State: Released
